theseus_gui/src-tauri/src/api/mod.rs

```diff
@@ -33,6 +32,10 @@ pub enum TheseusSerializableError {
 
     #[error("IO error: {0}")]
     IO(#[from] std::io::Error),
+
+    #[cfg(target_os = "macos")]
+    #[error("Callback error: {0}")]
+    Callback(String),
```

Comment:
what is this error used for and why is it macos specific?

Reply:
For the mac os delegate code (the interface to the objc code). This happens if the callback to the delegate is incorrectly set, or doubly set, etc. I have it only compile on mac as its only used in the mac callback code (which is only compiled on mac), so it'd be unused code otherwise

I could change it to be an unwrap() or a clippy ignored variant if you like